longhorndev/dev.go

```diff
@@ -404,16 +404,45 @@ func (d *LonghornDevice) Expand(size int64) error {
 	d.Lock()
 	defer d.Unlock()
 
-	if d.scsiDevice != nil {
-		return fmt.Errorf("cannot expand the device %v to size %v since the frontend %v is already up", d.name, size, d.frontend)
-	}
-
 	if d.size > size {
 		return fmt.Errorf("device %v: cannot expand the device from size %v to a smaller size %v", d.name, d.size, size)
 	} else if d.size == size {
 		return nil
 	}
 	d.size = size
 
+	if d.scsiDevice == nil {
+		return nil
+	}
+	if err := d.scsiDevice.UpdateScsiBackingStore("longhorn", fmt.Sprintf("size=%v", d.size)); err != nil {
+		return err
+	}
+
+	switch d.frontend {
+	case types.FrontendTGTBlockDev:
+		logrus.Infof("device %v: Expanding frontend %v target %v", d.name, d.frontend, d.scsiDevice.Target)
+		if err := d.scsiDevice.ExpandTarget(size); err != nil {
```

Comment:
Do we need a lock protection during expansion?

Reply:
There is already a lock at the beginning of this function. Do we need anything else?

Reply:
Ah, I see. It is fine.